src/main/java/io/cdap/plugin/sap/OData2Client.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.sap;
+
+import com.google.common.base.Strings;
+import io.cdap.plugin.sap.exception.ODataException;
+import org.apache.olingo.odata2.api.edm.Edm;
+import org.apache.olingo.odata2.api.edm.EdmEntitySet;
+import org.apache.olingo.odata2.api.edm.EdmEntityType;
+import org.apache.olingo.odata2.api.edm.EdmException;
+import org.apache.olingo.odata2.api.ep.EntityProvider;
+import org.apache.olingo.odata2.api.ep.EntityProviderException;
+import org.apache.olingo.odata2.api.ep.EntityProviderReadProperties;
+import org.apache.olingo.odata2.api.ep.feed.ODataFeed;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.List;
+import javax.annotation.Nullable;
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * Provides handy methods to consume OData v2 services using Apache Olingo v2 as Client Library.
+ */
+public class OData2Client {
+
+  private static final String METADATA = "$metadata";
+  private static final String SEPARATOR = "/";
+  private static final String QUERY_SEPARATOR = "?";
+
+  private String rootUrl;
+  private String username;
+  private String password;
+
+  /**
+   * @param rootUrl  URL of the OData service. The URL must end with an external service name
+   *                 (e.g., http://eccsvrname:8000/sap/opu/odata/sap/zgw100_dd02l_so_srv/).
```

Comment:
is it always expected to be of this form (http://[host]:[port]/sap/opu/odata/sap/[service name])? If so, do we validate that anywhere?

Reply:
All examples use this form but it seems that it's not documented by SAP as a standard form. Unfortunately, I can not find this in docs.